api/src/main/java/org/openmrs/module/metrics/api/db/MetricsDAO.java

```diff
@@ -12,7 +12,22 @@
 	
 	Integer getEncounterObjectsByGivenDateRangeAndType(Date startRange, Date endRange, String encounterType);
 	
-	Integer getNewPatientsObjectsByGivenDateRange(Date startRange, Date endRange);
+	Integer getNewObjectsByGivenDateRangeAndCategory(Date startRange, Date endRange, String category, String action);
 	
 	Map<String, Integer> getEncounterObjectTypesCountByGivenDateRange(Date startRange, Date endRange);
+	
+	Integer fetchTotalOpenMRSObjectCountTodayByEventAction(@SuppressWarnings("rawtypes") Class clazz, String actio);
```

Comment:
Either of these:

```suggestion
	Integer fetchTotalOpenMRSObjectCountTodayByEventAction(Class<?> clazz, String action);
```

```suggestion
	Integer fetchTotalOpenMRSObjectCountTodayByEventAction(Class<? extends OpenmrsObject> clazz, String action);
```

That goes for the rest of these.

Reply:
Generally speaking, instead of having methods for every possible period, wouldn't it be better to encode the period as a parameter, e.g.,

```java
int fetchOpenmrsObjectCountByActionForPeriod(Class<? extends OpenmrsObject> clazz, String action, Period period)
```

So this could be called like:

```java
fetchOpenmrsObjectCountByActionForPeriod(Patient.class, "CREATED", Period.of(Duration.ofDays(7)));
```

Reply:
I'd probably return an `int` rather than an `Integer`. Also, can `action` be turned into an Enum that enumerates the actions we capture rather than relying on String values or is there a structural reason we need Strings?

Reply:
no we can go with the enum